Meteor 3 compatibility release

StorytellerCZ opened the pull request:
Based on #59, this PR takes the next step and makes things compatible with Meteor 3.

## TODOs left

- [x] headless tests and CI fixed
- [x] no version conflicts with Meteor >=3.0.1 && < 3.4
- [x] no version conflicts with Meteor >= 3.4
- [ ] fix ddp queueing (see https://github.com/Meteor-Community-Packages/meteor-timesync/pull/60#issuecomment-2774954094)

StorytellerCZ committed:
Initial work for Meteor 3 compatibility

StorytellerCZ committed:
Published mizzao:timesync@1.0.0-beta.0.

StorytellerCZ commented:
I'm wondering if mtest is compatible with Meteor 3. I see no reason not to, but the cancellation is strange.

wreiske commented:
Looking forward to this! I tried getting the latest versions of both running with Meteor 3.0.2 but ran into this...

```
meteor add mizzao:user-status@2.0.0-beta.0
mizzao:user-status with version constraint 2.0.0-beta.0 has already been added.

wreiske@Wills-MacBook-Pro bluehive-ai % meteor add mizzao:timesync@1.0.0-beta.0
 => Errors while adding packages:             
                                              
While selecting package versions:
error: Conflict: Constraint mizzao:timesync@0.5.4 is not satisfied by mizzao:timesync 1.0.0-beta.0.
Constraints on package "mizzao:timesync":
* mizzao:timesync@1.0.0-beta.0 <- top level
* mizzao:timesync@0.5.4 <- mizzao:user-status 2.0.0-beta.0
```

StorytellerCZ commented:
@wreiske I'll release a new rc for `mizzao:user-status` to fix this shortly.

alisnic commented:
After upgrading to Meteor 3.0.4 and to timesync 1.0.0 beta, the _timeSync method is called very often:

![Screenshot 2025-01-10 at 11 09 42](https://github.com/user-attachments/assets/f62c3d39-9dfe-4e92-9fc5-bb9dc40878ca)

These are stats for 7 active containers on galaxy and about 700 connections

StorytellerCZ commented:
That is a concern, but at first look I don't see any change that should cause an increase. 🤔

alisnic commented:
@StorytellerCZ my take is that timesync does not account that ddp methods are queued, and if timeSync calls are blocked by other method calls it will just re-issue timeSync calls. I think this whole DDP thing needs to be reverted

jankapunkt commented:
@StorytellerCZ this is blocking because mtest is not meteor 3 compatible

jankapunkt commented:
related https://github.com/zodern/mtest/issues/2

BastienRodz committed:
fix(cordova): on mobile we need to force ddp to avoid versions errors

allenfuller commented:
Anything those of us in the community can do to help? Just updated to Meteor 3.4 and unable to update this package.
```
Newer versions of the following indirect dependencies are available:
 * mizzao:timesync 0.5.5 (0.6.0 is available) 
```

jankapunkt commented:
Hey @allenfuller I also found the update to 3.4 caused a lot of dependency issues. I think we need to bump core versions to include 3.4.

jankapunkt committed:
published 1.0.0-beta.1

jankapunkt commented:
I published a new version `1.0.0-beta.1` which you can try to install. I also noticed the issue @alisnic reported is not really resolved which is why I will assign myself to this PR to make this work and publish the package.

jankapunkt committed:
tests: move to Mocha based tests

jankapunkt committed:
fix(ci): use correct Meteor version args

jankapunkt commented:
@StorytellerCZ  I would like to move biome with this

jankapunkt committed:
deps: update npm deps

harryadel committed:
Merge branch 'feature/meteor3' into fix/cordova-meteor3

BastienRodz committed:
test(cordova): regression tests for Cordova DDP transport selection

Add two Mocha tests under a 'transport selection' describe block:

- forces DDP on Meteor.isCordova=true — asserts Meteor.callAsync('_timeSync')
  is used and globalThis.fetch to the sync URL is not. Reproduces the
  scenario that motivated the fix (Cordova client with useDDP still false).

- uses HTTP on a plain browser — guards the else fetch(...) branch of
  updateOffset() against accidental deletion, proving the fix does not
  regress the web.browser path.

Both tests install spies on Meteor.callAsync and globalThis.fetch, drive
a TimeSync.resync(), and poll with simplePoll rather than a fixed timeout
to stay reliable under CI load.

BastienRodz committed:
test(cordova): spy HTTP transport via PerformanceObserver

The previous fetch spy installed itself on globalThis.fetch, but
timesync-client.js imports fetch from meteor/fetch as a module binding
at load time. Replacing globalThis.fetch had no effect, so the browser
regression test could never observe an HTTP sync and timed out.

Switch to PerformanceObserver on 'resource' entries, which captures
any fetch() to /_timesync regardless of how the caller references it.

harryadel committed:
Merge pull request #61 from BastienRodz/fix/cordova-meteor3

harryadel committed:
Merge branch 'master' into feature/meteor3

harryadel committed:
Publish 1.0.0-beta.2

harryadel commented:
Published `1.0.0-beta.2` thanks to @BastienRodz contribution.

harryadel committed:
Prepare 1.0.0 release
